snyk/snyk
# feat: add the error string code to analytics

#### What does this PR do?
This PR defines error string codes for IaC errors, to improve readability in analytics.

#### Where should the reviewer start?


#### How should this be manually tested?

1. Run `npm run build`
2. Run a failing scan: `node ./dist/cli iac test test/fixtures/iac/kubernetes/pod-invalid.yaml --experimental -d`
3. Notice the `error-str-code` being included under the metadata field.

#### Any background context you want to provide?


#### What are the relevant tickets?
- [Jira ticket CC-786](https://snyksec.atlassian.net/browse/CC-786)

#### Screenshots
![Screenshot 2021-05-07 at 11 48 02](https://user-images.githubusercontent.com/81559517/117440484-01e2e180-af2c-11eb-8e16-245256f0e424.png)
